src/cobald/composite/factory.py

```diff
@@ -12,23 +12,27 @@ class FactoryPool(CompositePool):
     """
     Composition that adds and removes pools to satisfy demand
 
-    :param factory: a callable that produces a single new :py:class:`~.Pool` on each invocation
+    :param factory: a callable that produces a new :py:class:`~.Pool`
     :param interval: how often to adjust the number of children
 
     Adjustment uses two extensions that children must respond to adequately:
 
-    * When spawned via ``factory()``, children shall already be set to their expected ``demand``.
-    * When disabled via ``demand=0``, children shall shut down and free any resources and tasks.
+    * When spawned via ``factory()``, children shall already be
+      set to their expected ``demand``.
+    * When disabled via ``demand=0``, children shall shut down
+      and free any resources and tasks.
 
     Once spawned, children are free to adjust their demand if required.
     A child may disable itself permanently by setting ``demand=0``.
-    The :py:class:`FactoryPool` re-inspects child demand before spawning or disabling children.
-
-    Note that both ``allocation`` and ``utilisation`` are computed only from children which
-    satisfy ``supply > 0`` and ``demand > 0``.
-    All other machines are considered as not fully functional -
-    and thus as representative due to not aiming for ideal usage.
-    The ``children``, ``demand`` and ``supply`` reflect all available resources, however.
+    The :py:class:`FactoryPool` re-inspects child demand before spawning
+    or disabling children.
+
+    Note that both ``allocation`` and ``utilisation`` are computed only
+    from children which satisfy ``supply > 0`` and ``demand > 0``.
+    All other children are considered as not fully functional -
```

Comment:
```suggestion
    We consider all other children as not fully functional --
```

What is the purpose of this? If `demand = 0` shouldn't you still consider `allocation` and `utilisation`? 
However, this also depends on the way a child is shut down when `demand` is set to `0` (is it either killed or can finish smoothly).

Reply:
The idea is that during shutdown the resource is probably not used optimally anymore - e.g. it gets drained and thus has lower utilisation/allocation. Thus, it gets excluded from the estimation for how well the resource works.

There are basically two cases:
* Child shuts down immediately. We don't care about the efficiency because the child is gone.
* Child shuts down slowly. We don't care about the efficiency because the child is not working optimally.

Should this reasoning be part of the documentation?

Reply:
I still see an issue here. When the `demand` is `0` already this means a bad utilisation does not harm as `demand` cannot be decreased anymore. However, globally, we are still interested in approximating optimal utilisation and allocation. And by excluding not well used resources as we decided to disintegrate them is somehow artificially increasing the goodness of our approach. I think also the resource usage for booting and shutdown must be considered to estimate the actual costs. So as long as we are running the resource and it is correctly integrated, I would consider its statistics.

Reply:
I recommend to move this to a separate issue. Changing the docstring was simply due to line length, not content. Even if we do change it, it should be a separate change.

Reply:
Opened an issue for this as #35.